include/echoFunctions.php

```diff
@@ -0,0 +1,216 @@
+<?php
+
+
+function echoHeader($Title, $PageName) {
+
+	?>
+	<script src="/include/jquery.js"></script>
+	<?php
+
+	echo "
+		<head>
+			<title>$Title</title>
+			<link rel='stylesheet' href='style.css'/>
+			<link rel='stylesheet' href='https://fonts.googleapis.com/css?family=Quicksand'>
+			<link href='https://fonts.googleapis.com/css?family=Oxygen' rel='stylesheet'>
+		</head>
+
+";
+?>
+
+<script type='text/javascript'>
+
+
+function intakeVote(eventID, userID, direction){
+
+//both buttons haven't been pressed
+	if(!document.getElementById("upVoteButton_"+eventID).classList.contains('voted') && !document.getElementById("downVoteButton_"+eventID).classList.contains('voted')){
+		if(direction == "up"){
+			jsUpVote(eventID, userID, direction);
+		}
+		else{ //direction is down
+			jsDownVote(eventID, userID, direction);
+		}
+	}
+
+//if vote exists
+
+	//the downvote has been pressed but not the upvote
+	else if(!document.getElementById("upVoteButton_"+eventID).classList.contains('voted') && document.getElementById("downVoteButton_"+eventID).classList.contains('voted')){
+		if(direction == "up"){
+			undoDownVote(eventID, userID, direction, "double");
+			jsUpVote(eventID, userID, direction);
+		}
+		else{ //direction is down
+			undoDownVote(eventID, userID, direction, null);
+		}
+	}
+
+	//the upvote has been pressed but not the downvote
+	else{
+		if(direction == "up"){
+			undoUpVote(eventID, userID, direction, null);
+		}
+		else{ //direction is down
+			undoUpVote(eventID, userID, direction, "double");
+			jsDownVote(eventID, userID, direction);
+		}
+
+	}
+}
+
+
+
+function jsUpVote(eventID, userID, direction){
+	//update UI so gets unvoted,
+	//ajax request to unvote this in the server.
+	document.getElementById("upVoteButton_"+eventID).classList.add('voted');
+
+	//gets the number
+	var insideTheDiv = document.getElementById("eventWrapper_"+eventID).innerHTML;
+	var pureNumber = parseInt(insideTheDiv);
+	console.log("before you voted = " +pureNumber);
+
+	//changes the number in the UI
+	document.getElementById("eventWrapper_"+eventID).innerHTML = pureNumber+1;
+	console.log("after you voted = " +(pureNumber+1));
+	// console.log(eventID, userID, direction);
+
+	//call ajax
+	$.post({ url:'/ajax/ajaxVote.php', data:{eventID, userID, direction}});
+
+}
+
+
+function jsDownVote(eventID, userID, direction){
+
+	document.getElementById("downVoteButton_"+eventID).classList.add('voted');
+
+	//gets the number
+	var insideTheDiv = document.getElementById("eventWrapper_"+eventID).innerHTML;
+	var pureNumber = parseInt(insideTheDiv);
+	console.log("before you voted = " +pureNumber);
+
+	//changes the number in the UI
+	document.getElementById("eventWrapper_"+eventID).innerHTML = pureNumber-1;
+	console.log("after you voted = " +(pureNumber-1));
+
+	//call ajax
+	$.post({ url:'/ajax/ajaxVote.php', data:{eventID, userID, direction}});
+
+}
+
+function undoUpVote(eventID, userID, direction, ifUndoAndVote){
```

Comment:
These functions look fine, but if you refactor some of the stuff I mentioned above (making this more abstract, etc.) it's possible that these could be improved to match.

Reply:
My only comment is that maybe undoing and voting should be two separate functions rather than one. The undo doesn't need to send anything to the server (when the server gets the vote request, it can manage the upvote itself). So you'd call `undoUpVote` followed by `jsDownVote` so you don't need any voting logic in here (nor do you need the direction)